106-auto-rebalance-on-imbalanced-clusters.md

```diff
@@ -0,0 +1,583 @@
+If the user runs a `manual` rebalance, we will empty the `anomalyList` in those cases too.
+If some other anomalies are detected again after the rebalance then they will be updated in the ConfigMap by the notifier and another rebalance will take place
+
+The ConfigMap will also have an option field called `unfixedAnomaly` list which would be created only when an unfixable anomaly is detected.
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: my-cluster-goal-violation-map
+data:
+  last_anomaly_detection_timestamp: <when last anomaly detection was run>
+  rebalance_finished_timestamp: <when the rebalance finished>
+  anomalyList: |
+    - <anomaly-id-1>
+    - <anomaly-id-2>
+  unfixableGoals: |
+    - <unfixed-goal-1>
+    - <unfixed-goal-2>  
+# ...
+```
+The operator will use the `unfixableGoals` list to check if there are any unfixable goals detected or not. 
+If the list is not empty then the operator would create a `warning` condition based on the unfixable goal and add it to the `Kafka` CR status.
+
+The `last_anomaly_detection_timestamp` corresponds to when the last anomaly was detected and will be updated every time a new anomaly is detected.
+
+The `rebalance_finished_timestamp` refers to the time when the last running rebalance finished and will be updated every time when a rebalance is finished.
+The rebalance can be either auto-rebalance or manual rebalance.
+
+The created ConfigMap will persist when self-healing is enabled and will be deleted if the user decides to disable self-healing.
+This is done to keep track of the timestamps which will help us to avoid unnecessary rebalances.
+
+The `StrimziCruiseControlNotifier` will be kept in a separate repository under the Strimzi organization.
+
+#### What happens if same anomaly type is detected while the auto-rebalance/manual rebalance is happening
+
+The anomaly detector continues to run, even when a rebalance has been triggered.
+Therefore, it is possible that it may detect an anomaly that is currently in the process of being fixed.
+If we logged the anomaly in the ConfigMap, we may then trigger another rebalance for an anomaly that was fixed by the ongoing rebalance.
+This could make operator running multiple rebalances for the anomaly and therefore creating a never ending rebalance loop.
+
+To tackle this problem we can make use of the `last_anomaly_detection_timestamp` and `rebalance_finished_timestamp` timestamps
+Everytime an anomaly of the same type is detected we will treat its detection time as the `last_anomaly_detection_time` and update the value of it in the `last_anomaly_detection_time` field of the ConfigMap.
+Now if the anomaly detection happened during the rebalance, the `ConfigMap` will be updated with the anomalies as well as the `last_anomaly_detection_time` will also be updated with last anomaly detection time.
+We will track the rebalance finish time in ConfigMap under the `rebalance_finished_timestamp` field.
```

Comment:
You have basically already said this (regarding both timestamps) in the section above. I don't think you need to spell it out again.